extensions/shared/external-media/external-media-modal.js

```diff
@@ -0,0 +1,288 @@
+/**
+ * External dependencies
+ */
+import { uniqBy } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { compose } from '@wordpress/compose';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { Component } from '@wordpress/element';
+import { withNotices, Modal } from '@wordpress/components';
+import { UP, DOWN, LEFT, RIGHT } from '@wordpress/keycodes';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { PATH_RECENT } from './constants';
+import MediaItem from './media-browser/media-item';
+
+const MODAL_NAME = 'edit-post/external-media';
+
+const CopyingMedia = ( { items } ) => {
+	const classname =
+		items.length === 1
+			? 'jetpack-external-media-browser__single'
+			: 'jetpack-external-media-browser';
+
+	return (
+		<div className={ classname }>
+			<div className="jetpack-external-media-browser__media">
+				{ items.map( item => (
+					<MediaItem item={ item } key={ item.ID } isSelected isCopying />
+				) ) }
+			</div>
+		</div>
+	);
+};
+
+class ExternalMediaModal extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			media: [],
+			nextHandle: false,
+			isLoading: false,
+			isCopying: null,
+			requiresAuth: false,
+			path: { ID: PATH_RECENT },
+		};
+
+		this.arrowKeysPropagationStopped = false;
+	}
+
+	componentDidUpdate() {
+		this.handleArrowKeysPropagation();
```

Comment:
Why is this added on update instead of on mount?

Reply:
The modal is mounted when the block is added to the editor canvas and unmounted when the block is removed. The listener element is not available on mount because it is rendered only when the modal is opened.

Reply:
Decided to restore the original implementation - now the handler is attached on mount :)
See https://github.com/Automattic/jetpack/pull/16055/commits/bf3eb0f2bed2bdcfd8ac81f05c833715a6ac2b85.